### PR TITLE
Sysusers, tmpfiles, linx-cleanup, linx-genkey, remove config.ini

### PR DESCRIPTION
**Pull request by crabvk**

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -3,32 +3,50 @@
 
 pkgname=linx-server
 pkgver=2.3.5
-pkgrel=1
-pkgdesc='Self-hosted file/code/media sharing website '
-arch=('i686' 'x86_64' 'arm')
```
Comment (Thor77) [inline PKGBUILD:8]: Why are you removing those architectures?

File: PKGBUILD [hunk 2]
```diff
@@ -3,32 +3,50 @@
...
+pkgrel=2
+pkgdesc='Self-hosted file/code/media sharing website'
+arch=('x86_64')
 url='https://github.com/andreimarcu/linx-server'
 license=('GPL3')
-install=linx-server.install
```
Comment (Thor77) [inline PKGBUILD:11]: I agree that using sysusers/tmpfiles is better than creating the user/group/dir manually, but you still have to call `systemd-sysusers/tmpfiles` somewhere.

File: linx-server.service [hunk 3]
```diff
@@ -5,7 +5,10 @@ After=network.target
 [Service]
 User=linx
 Group=linx
-ExecStart=/usr/bin/linx-server -config /etc/webapps/linx-server/config.ini
+ExecStart=/usr/bin/linx-server -config /etc/linx-server/config.ini -allowMissingConfig
+WorkingDirectory=/var/lib/linx
+Restart=always
```
Comment (Thor77) [inline linx-server.service:10]:
```suggestion
Restart=on-failure
```

File: linx-server.service [hunk 4]
```diff
@@ -5,7 +5,10 @@ After=network.target
...
+RestartSec=2s
```
Comment (Thor77) [inline linx-server.service:11]: Why exactly 2s? I guess the default is fine.

**Review by Thor77 — CHANGES_REQUESTED** [review]
Thanks for your PR!
I left some comments, though.
And I'm not really sure why `/etc/webapps/linx-server` is worse than `/etc/linx-server` as other packages are using the location as well and the [Web application package guidelines](https://wiki.archlinux.org/index.php/Web_application_package_guidelines) suggest it as well.

File: PKGBUILD (hunk 1 above)
Comment (crabvk) [reply PKGBUILD:8]: Sorry, I misread the PKGBUILD docs.

File: PKGBUILD (hunk 2 above)
Comment (crabvk) [reply PKGBUILD:11]: In `package()` those file copied to `/usr/lib/{sysusers.d,tmpfiles.d}` and this triggers pacman hooks from `/usr/share/libalpm/hooks/{20-systemd-sysusers.hook,30-systemd-tmpfiles.hook}`.

File: linx-server.service (hunk 3 above)
Comment (crabvk) [reply linx-server.service:10]: ok

File: linx-server.service (hunk 4 above)
Comment (crabvk) [reply linx-server.service:11]: The default is 0.01s, isn't it too often? I just took `always` and `2s` from gitea :)

**Comment by crabvk** [comment]
You're right about /etc/webapps, lots of packages use it (but not gitea for some reason), I didn't know.

File: PKGBUILD (hunk 2 above)
Comment (Thor77) [reply PKGBUILD:11]: Oh, I didn't know about those hooks, nvm then.

File: linx-server.service (hunk 4 above)
Comment (Thor77) [reply linx-server.service:11]: Alright, we can also keep it like this, 2s just seemd a little bit random to me :)

**Comment by crabvk** [comment]
I asked guys from #archlinux (irc channel on freenode.net) about relevance of /etc/webapps directory. It turned out that FHS does not define anything web related. Here's the talk

> FinnMertens: What is web application? Things like gitlab (ruby) or wordpress (php) are web applications. But what if an app is in go lang with all assets bundled inside the binary?
> Namarrgon: how many parts it has is not relevant in the definition of a "web app"
> FinnMertens: Namarrgon: idk for sure, but a client could be curl (without a browser)
> FinnMertens: Namarrgon: https://wiki.archlinux.org/index.php/Web_application_package_guidelines
> Namarrgon: FinnMertens: sure, the definition is vague enough but what are you actually trying to do with that knowledge?
> FinnMertens: Why should I prefer /etc/webapps/linx-server/config.ini over /etc/linx-server/config.ini ?
> mkoskar: /etc/webapps? eww really?
> Namarrgon: i question the relevance of the information of that page
> Namarrgon: nobody is going to complain about the latter of the two paths you suggest
> Namarrgon: "The layout follows the FHS." does it really define where "webapp" configs and data go?
> FinnMertens: Namarrgon: I can't find anything about "web" in FHS
> Namarrgon: **forget about that page**
> FinnMertens: Namarrgon: :D thanks
> mkoskar: FinnMertens: webbaps subdir is eww, it creates exactly the problem you have right now, flat structure is just fine, and what if you have a package that webapp-y bits and other bits?
> mkoskar: *that contains

**Comment by Thor77** [comment]
Alright, we can also remove the `webapps`-prefix.
There should be some kind of migration-notice, though to inform existing users (if there are any) of the change.

**Comment by crabvk** [comment]
I've added a notice for the case when previous package version is lower than 2.3.5-2. Please have a look.

**Comment by Thor77** [comment]
Merged with some changes (readded config.ini, modified login shell of linx user, set data dir to /var/lib/linx-server).
Thanks!

**Pull request closed without merging**